solutions/0000-0099/0036-valid-sudoku-easy.md

```diff
@@ -1,5 +1,168 @@
+```
+
+**Constraints:**
+
+* `board.length == 9`
+* `board[i].length == 9`
+* `board[i][j]` is a digit `1-9` or `'.'`.
+
+## Approach 1: HashSet
+
+Every value in the row or column or `3 x 3` block, the value is unique, With the help of this simple logic, add the Numbers in the HashSet, Add `num` value for `m`th row and for `n`th column and the same for `3 x 3` matrix. 
+
+If the same value repeat again for the same row or column or `3 x 3` block, then the given sudoku is not valid, otherwise is valid.
+
+Time Complexity: O(m * n), where m - # of rows, n - # of cols
+Space complexity: O(1), since the values are fixed for `9 x 9` board
+
+<Tabs>
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@vigneshshiv"/>
+
+```java
+class Solution {
+    public boolean isValidSudoku(char[][] board) {
+        Set<String> seen = new HashSet<>();
+        for (int r = 0; r < 9; r++) {
+            for (int c = 0; c < 9; c++) {
+                int num = board[r][c];
+                if (num != '.') {
+                    if (!seen.add(num + " in row " + r) || !seen.add(num + " in column " + c)
+                            || !seen.add(num + " in block " + (r / 3) + "-" + (c / 3))) {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Standard
+
+Iterate over rows and columns and `3 x 3` block to check if the number is repeats. 
+
+Time complexity: O(9^n^2)
+Space complexity: O(n)
+
+<Tabs>
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@vigneshshiv"/>
+
+```java
+class Solution {
+    public boolean isValidSudoku(char[][] board) {
+        for (int row = 0; row < board.length; row++) {
+            for (int col = 0; col < board[0].length; col++) {
+                if (board[row][col] != '.' && !isValid(board, row, col, board[row][col])) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+    
+    public boolean isValid(char[][] board, int row, int col, int num) {
+        // Check the row, from col 0 to 8
+        for (int i = 0; i < board.length; i++) {
+            if (i != col && board[row][i] == num) {
+                return false;
+            }
+        }
+        // Check the column, from row 0 to row 8
+        for (int i = 0; i < board[0].length; i++) {
+            if (i != row && board[i][col] == num) {
+                return false;
+            }
+        }
+        //
+        int sqrt = (int) Math.sqrt(board.length);
+        int rowStart = row - row % sqrt;
+        int colStart = col - col % sqrt;
+        for (int r = rowStart; r < rowStart + sqrt; r++) {
+            for (int c = colStart; c < colStart + sqrt; c++) {
+                if (row != r && col != c && board[r][c] == num) {
+                    return false;
+                }
+            }
+        }
```

Comment:
Need detailed explanations for beginners.

Reply:
Hello wingwong, 
The solution posted here are accepted in Leetcode for the problem and I do maintain a repo for this. Please take a look for this problem solution on this - https://github.com/vigneshshiv/leetcode/blob/main/code/java/backtracking/ValidSudoku.java

Sure, about the Time and Space complexity, I will change it and raise a PR.

The last one about, searching the value on 3 x 3 grid, this helps to find starting position on which row and on which column. I will add the explanation..